CMakeLists.txt

```diff
@@ -377,7 +387,8 @@ else()
 	message(WARNING, "Python 2 not found; enums will not be scanned.")
 endif()
 
-target_link_libraries(pioneer-lib PUBLIC lz4 fmt::fmt)
+target_link_libraries(pioneer-lib PUBLIC lz4 fmt::fmt $<$<BOOL:${BUILD_WITH_OPENAL}>:OpenAL::OpenAL>)
+target_compile_definitions(pioneer-lib PUBLIC $<$<BOOL:${BUILD_WITH_OPENAL}>:PI_BUILD_WITH_OPENAL>)
```

Comment:
Because this is applied as a public compilation definition, changing the option `BUILD_WITH_OPENAL` triggers a full rebuild of the codebase. Instead, I'd recommend you use the file `buildopts.h` (generated via CMAKE) and `#include` that in only the appropriate source files.

Reply:
Nice catch, fixed.